src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,7 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        saveToAddressBookIfMutating(command.isMutating());
```

Comment:
Since you changed the behavior, you should update the header comment accordingly.